pkg/kubectl/cmd/util/editor/editoptions.go

```diff
@@ -320,11 +323,33 @@ func (o *EditOptions) Run() error {
 
 	switch o.EditMode {
 	// If doing normal edit we cannot use Visit because we need to edit a list for convenience. Ref: #20519
-	case NormalEditMode:
+	case NormalEditMode, ApplyEditMode:
 		infos, err := o.OriginalResult.Infos()
 		if err != nil {
 			return err
 		}
+		if o.EditMode == ApplyEditMode {
```

Comment:
I would break `case NormalEditMode, ApplyEditMode` into 2 seperate cases instead of using another if conditioning in the case statement.
```
case NormalEditMode:

case ApplyEditMode:

```